galasa-inttests-parent/dev.galasa.inttests/src/main/java/dev/galasa/inttests/docker/AbstractDocker.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.inttests.docker;
+
+import org.apache.commons.logging.Log;
+
+import dev.galasa.core.manager.Logger;
+import dev.galasa.ipnetwork.ICommandShell;
+
+public abstract class AbstractDocker {
+	
+	@Logger
+	private Log logger;
+	private ICommandShell shell;
+	
+	protected void setShell(ICommandShell shell) throws Exception {
+		this.shell = shell;
+	}
+	
+	protected void updatePackageManager(String command) throws Exception {
+		shell.issueCommand(command);
+	}
+	
+	protected boolean isDockerInstalled() throws Exception {
+		logger.info("Checking Docker is installed.");
+		String res = shell.issueCommand("docker -v");	
+		if(res.contains("no such file or directory")) {
+			return false;
+		} else {
+			return true;
+		}
+	}
+	
+	protected void installDocker(String command) throws Exception {
+		logger.info("Installing Docker");
+		shell.issueCommand(command);
+	}
+	
+	protected boolean isDockerRunning(String command) throws Exception {
```

Comment:
think all these docker methods can be in a single method.   will make for easier reading